### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by jason-fox**
* Update `package.json` to use latest `dependencies` and add `devDependencies`.
* Remove old config files
* Update **travis-ci** - linter dependencies are now automatically installed as part `npm install`
* Let **prettier** reformat the code.
* Correct outstanding linting errors

### CODE REVIEW

**Comment by chicco785** [comment]
@jason-fox while i understand the purpose, i am not sure i am in favour of autofixing code. 
personally, i prefer to have a failed build, and see what i have to change.

feelings?

File: validator/libs/checks.js [hunk 1]
```diff
@@ -1,68 +1,67 @@
...
+      const stat = fs.lstatSync(fullPath);
       if (stat && stat.isDirectory() &&
            !conf.nconf.get('dmv:ignoreFolders')
              .includes(path.basename(fullPath)) &&
            !conf.nconf.get('dmv:docFolders')
              .includes(path.basename(fullPath)) &&
            !conf.nconf.get('dmv:externalSchemaFolders')
              .includes(path.basename(fullPath)))
-      folderCounter++;
+      {folderCounter++;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 2]
```diff
@@ -1,68 +1,67 @@
...
       debug("*containsModelFolders* - folders found: "+ folderCounter);
     } catch (err) {
       console.log('***ERROR*** ' + err);
-      if (conf.failErrors) throw new Error(err);
+      if (conf.failErrors) {throw new Error(err);}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 3]
```diff
@@ -1,68 +1,67 @@
...
     }
   });
-  if (folderCounter) return true;
-  else return false;
+  if (folderCounter) {return true;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 4]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
-
+        console.log('***ERROR*** ' + err);
       }
     });
 
     if (!check && msg.addWarning(fullPath, 'does not include a documentation ' +
           'file named spec.md or introduction.md') && conf.failWarnings)
-      throw new Error('Fail on Warnings: ' +
-        JSON.stringify(msg.warnings, null, '\t'));
+      {throw new Error('Fail on Warnings: ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 5]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     });
-    if (counter == 0)
-      msg.addWarning(fullPath, 'does not include a documentation folder');
+    if (counter === 0)
+      {msg.addWarning(fullPath, 'does not include a documentation folder');}
     if (conf.nconf.get('dmv:warningChecks').includes('docExist') &&
-        counter == 0)
-      if (msg.addWarning(fullPath, 'does not include a documentation ' +
+        counter === 0)
+      {if (msg.addWarning(fullPath, 'does not include a documentation ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 6]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
         counter++;
       } catch (err) {
-
+        console.log('***ERROR*** ' + err);
       }
     });
-    if (counter == 0)
-      msg.addWarning(fullPath, 'does not include a documentation folder');
+    if (counter === 0)
+      {msg.addWarning(fullPath, 'does not include a documentation folder');}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 7]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     if (conf.nconf.get('dmv:warningChecks').includes('docExist') &&
-        counter == 0)
-      if (msg.addWarning(fullPath, 'does not include a documentation ' +
+        counter === 0)
+      {if (msg.addWarning(fullPath, 'does not include a documentation ' +
            'file named spec.md or introduction.md') && conf.failWarnings)
-        throw new Error('Fail on Warnings: ' +
-          JSON.stringify(msg.warnings, null, '\t'));
+        {throw new Error('Fail on Warnings: ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 8]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
+          JSON.stringify(msg.warnings, null, '\t'));}}
    debug("*docFolderExist* - " + fullPath + ": "+ counter);
-   if (counter >0) return true;
-   else return false;
+   if (counter >0) {return true;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 9]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
 
   //check if a folder name is valid for a data model
-  modelNameValid: function(fullPath) {
-    var check = true;
-    if (fullPath.charAt(0) != fullPath.charAt(0).toUpperCase())
-      check = false;
+  modelNameValid(fullPath) {
+    let check = true;
+    if (fullPath.charAt(0) !== fullPath.charAt(0).toUpperCase())
+      {check = false;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 10]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     if (!check &&
         msg.addWarning(fullPath, 'Model folder names should start' +
         ' in capital letter') && conf.failWarnings)
-      throw new Error('Fail on Warnings: ' +
-        JSON.stringify(msg.warnings, null, '\t'));
+      {throw new Error('Fail on Warnings: ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 11]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
   },
 
   //check if a folder includes a README.md file
-  readmeExist: function(fullPath) {
-    var check = true;
+  readmeExist(fullPath) {
+    let check = true;
     if (!fileExists(fullPath, 'README.md'))
-      check = false;
+      {check = false;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 12]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
   },
 
   //check if a folder includes a schema file
-  schemaExist: function(fullPath) {
-    var check = true;
+  schemaExist(fullPath) {
+    let check = true;
     if (!fileExists(fullPath, '^schema\\.json'))
-      check = false;
+      {check = false;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 13]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     if (!check && !containsModelFolders(fullPath) &&
         msg.addWarning(fullPath, 'does not include a JSON Schema ' +
           'file schema.json') && conf.failWarnings)
-      throw new Error('Fail on Warnings: ' +
-        JSON.stringify(msg.warnings, null, '\t'));
+      {throw new Error('Fail on Warnings: ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 14]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
   },
 
   //check if a folder includes one or more example files
-  exampleExist: function(fullPath) {
-    var check = true;
+  exampleExist(fullPath) {
+    let check = true;
     if (!fileExists(fullPath, '^example(-\\d+)?\\.json'))
-      check = false;
+      {check = false;}
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 15]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     if (!check && !containsModelFolders(fullPath) &&
         msg.addWarning(fullPath, 'does not include a JSON Example file ' +
           'example(-\\d+)?\\.json') && conf.failWarnings)
-      throw new Error('Fail on Warnings: ' +
-        JSON.stringify(msg.warnings, null, '\t'));
+      {throw new Error('Fail on Warnings: ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 16]
```diff
@@ -234,28 +235,30 @@ module.exports = {
        msg.addError(fullPath, 'JSON Example ' + fileName + ' is not supported by ' +
          'contextBroker: ' +  JSON.stringify(err));
        if (conf.failErrors)
-         throw new Error('Fail on Error: JSON Example ' + fileName + ' is not supported by ' +
-         'contextBroker: ' +  JSON.stringify(err));
+         {throw new Error('Fail on Error: JSON Example ' + fileName + ' is not supported by ' +
```
Comment (chicco785) [inline validator/libs/checks.js]: not sure i consider this linting :)

File: validator/libs/message.js [hunk 17]
```diff
@@ -1,60 +1,60 @@
...
+const validSchemas = {};
+const validExamples = {};
+const supportedExamples = {};
+
+const addMessageToMap = function(modelPath, message, map) {
+  const rootModel = getRootModelName(modelPath);
+  const fullMessage = modelPath + ': ' + message;
   if (map[rootModel] != null)
-  map[rootModel].push(fullMessage);
+  {map[rootModel].push(fullMessage);}
```
Comment (chicco785) [inline validator/libs/message.js]: not sure i consider this linting :)

File: validator/libs/message.js [hunk 18]
```diff
@@ -1,60 +1,60 @@
...
   else
-  map[rootModel] = [fullMessage];
+  {map[rootModel] = [fullMessage];}
```
Comment (chicco785) [inline validator/libs/message.js]: not sure i consider this linting :)

File: validator/libs/message.js [hunk 19]
```diff
@@ -1,60 +1,60 @@
...
 };
 
 //given a path, retrieve the name of the root model
-var getRootModelName = function(fullPath) {
-  var index = fullPath.indexOf(path.sep);
-  if (index > 0) return fullPath.substring(0, index);
-  else return fullPath;
+const getRootModelName = function(fullPath) {
+  const index = fullPath.indexOf(path.sep);
+  if (index > 0) {return fullPath.substring(0, index);}
```
Comment (chicco785) [inline validator/libs/message.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 20]
```diff
@@ -1,36 +1,36 @@
...
 };
 
 //load  a list of files, supports regex
-var getFiles = function(args) {
-    var files = [];
-    if (Array.isArray(args)) args.forEach(_getFiles);
-    else _getFiles(args);
+const getFiles = function(args) {
+    let files = [];
+    if (Array.isArray(args)) {args.forEach(_getFiles);}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 21]
```diff
@@ -1,36 +1,36 @@
...
+    else {_getFiles(args);}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 22]
```diff
@@ -60,46 +61,46 @@ var openFile = function(filename, suffix) {
...
-        if (typeof validate == 'function') {
+        if (typeof validate === 'function') {
           debug("*compileSchema* - valid schema " +file+ " :" +validate);
           msg.addValidSchema(fullPath, 'Schema ' + file + ' is valid');
         } else {
           debug("*compileSchema* - invalid schema " +file+ " :" +validate);
           msg.addError(fullPath, 'Schema ' + file +
             ' failed to compile to a function');
-          if (conf.failErrors) throw new Error(validate.errors);
+          if (conf.failErrors) {throw new Error(validate.errors);}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 23]
```diff
@@ -60,46 +61,46 @@ var openFile = function(filename, suffix) {
...
         'if one or more schemas cannot be retrieved, ' +
         'try using remote validation (dmv:resolveRemoteSchemas=true), ' +
-        'check if \"dmv:loadModelCommonSchemas\" is enabled ' +
+        'check if "dmv:loadModelCommonSchemas" is enabled ' +
         ' (if missing schemas are FIWARE common schemas) ' +
-        'or store third party schemas in the \"externalSchema\" folder: ' +
+        'or store third party schemas in the "externalSchema" folder: ' +
         err.message);
-      if (conf.failErrors) throw new Error(err.message);
+      if (conf.failErrors) {throw new Error(err.message);}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 24]
```diff
@@ -60,46 +61,46 @@ var openFile = function(filename, suffix) {
...
 
   //validate examples agains a compiled schema
-  validateExamples: function(fullPath, validate) {
-    var files = getFiles(fullPath + path.sep + 'example*.json');
-    if (typeof validate != 'function')
-    if (msg.addError(fullPath, 'Examples cannot be validated since ' +
+  validateExamples(fullPath, validate) {
+    const files = getFiles(fullPath + path.sep + 'example*.json');
+    if (typeof validate !== 'function')
+    {if (msg.addError(fullPath, 'Examples cannot be validated since ' +
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 25]
```diff
@@ -60,46 +61,46 @@ var openFile = function(filename, suffix) {
...
          'validation function cannot be computed. Probably not all schemas ' +
          'can be resolved correctly (check schema errors)') && conf.failErrors)
-      throw new Error('Fail on Error:' +
-        JSON.stringify(msg.errors, null, '\t'));
+      {throw new Error('Fail on Error:' +
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 26]
```diff
@@ -108,41 +109,41 @@ module.exports = {
...
+        if (typeof validate !== 'function') {
          debug("*validateExamples* - " + fileName +
            " cannot be validated");
          msg.addError(fullPath, 'Example ' + fileName +
             ' cannot be validated: ' +
             JSON.stringify(validate.errors, null));
           if (conf.failErrors)
-            throw new Error('Fail on Error:' +
-              JSON.stringify(msg.errors, null, '\t'));
+            {throw new Error('Fail on Error:' +
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 27]
```diff
@@ -108,41 +109,41 @@ module.exports = {
...
+              JSON.stringify(msg.errors, null, '\t'));}
         }
-        var validExample = validate(data);
+        const validExample = validate(data);
         debug("*validateExamples* - " +fileName+ " validity : " +validExample);
         if (validExample)
-          msg.addValidExample(fullPath, fileName + ' is valid');
+          {msg.addValidExample(fullPath, fileName + ' is valid');}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 28]
```diff
@@ -108,41 +109,41 @@ module.exports = {
...
         else {
           msg.addError(fullPath, 'Example ' + fileName + ' is invalid: ' +
             JSON.stringify(validate.errors, null));
-          if (conf.failErrors) throw new Error('Fail on Error:' +
-            JSON.stringify(msg.errors, null, '\t'));
+          if (conf.failErrors) {throw new Error('Fail on Error:' +
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 29]
```diff
@@ -108,41 +109,41 @@ module.exports = {
...
+            JSON.stringify(msg.errors, null, '\t'));}
         }
       });
     } catch (err) {
       if (conf.failErrors)
-        throw new Error('Fail on Error:' +
-          JSON.stringify(msg.errors, null, '\t'));
+        {throw new Error('Fail on Error:' +
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 30]
```diff
@@ -108,41 +109,41 @@ module.exports = {
...
 
-  addUniqueToArray: function(array1, array2) {
-    var result = Array.from(array1);
+  addUniqueToArray(array1, array2) {
+    const result = Array.from(array1);
     array2.forEach(function(item2) {
       if (!array1.includes(item2) &&
           !array1.includes(path.basename(item2)))
-        result.push(item2);
+        {result.push(item2);}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 31]
```diff
@@ -151,9 +152,9 @@ module.exports = {
   loadSchema: function loadSchema(uri, callback) {
     request(uri, call);
     debug("*loadSchema* - uri: " + uri);
-    var call = function(err, res, body) {
+    const call = function(err, res, body) {
       if (err || res.statusCode >= 400)
-      callback(err || new Error('Loading error: ' + res.statusCode));
+      {callback(err || new Error('Loading error: ' + res.statusCode));}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 32]
```diff
@@ -162,18 +163,18 @@ module.exports = {
...
   // load schemas local to FIWARE Data Model
   // (that should be named using *-schema.json pattern)
-  loadLocalSchemas: function(fullPath) {
-    var files;
-    if (fullPath != '.')
-      files = getFiles(fullPath + path.sep + '*-schema.json');
+  loadLocalSchemas(fullPath) {
+    let files;
+    if (fullPath !== '.')
+      {files = getFiles(fullPath + path.sep + '*-schema.json');}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/schema.js [hunk 33]
```diff
@@ -162,18 +163,18 @@ module.exports = {
...
     else
-      files = getFiles('*-schema.json');
+      {files = getFiles('*-schema.json');}
```
Comment (chicco785) [inline validator/libs/schema.js]: not sure i consider this linting :)

File: validator/libs/checks.js [hunk 34]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
     conf.nconf.get('dmv:docFolders').forEach(function(value) {
       try {
         fs.lstatSync(path.join(fullPath, value)).isDirectory();
         if (fileExists(path.join(fullPath, value), 'spec.md') ||
             fileExists(path.join(fullPath, value), 'introduction.md'))
-         check = true;
+         {check = true;}
       } catch (err) {
-
+        console.log('***ERROR*** ' + err);
```
Comment (chicco785) [inline validator/libs/checks.js]: introducing the log, you have this errors in the console:
https://travis-ci.org/Fiware/dataModels/jobs/436145121#L560

but this is anyhow reported due to check= false in the final report:
https://travis-ci.org/Fiware/dataModels/jobs/436145121#L1123

File: validator/libs/checks.js [hunk 35]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
-    var counter = 0;
+  docFolderExist(fullPath) {
+    let counter = 0;
     conf.nconf.get('dmv:docFolders').forEach(function(value) {
       try {
         fs.lstatSync(path.join(fullPath, value)).isDirectory();
         counter++;
       } catch (err) {
-
+        console.log('***ERROR*** ' + err);
```
Comment (chicco785) [inline validator/libs/checks.js]: see above

**Comment by jason-fox** [comment]
@chicco785  - The philosophy behind the PR can be found [here](https://medium.freecodecamp.org/why-robots-should-format-our-code-159fd06d17f7)

The PR currently has a minimal config to show prettier standardizing code - the actual adding or positioning of `{` or what have you is not very relevant. The actual rules to apply can be configured of course.

The point is not that *your coding style is better than mine* or *my coding style is better than yours*,   it is that a robot applies a common *neutral* standard.

File: validator/libs/checks.js (hunk 34 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed - a5c41847

File: validator/libs/checks.js (hunk 35 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed a5c41847

**Comment by chicco785** [comment]
@jason-fox as said in general i prefer to have a failed build, because i don't comply, than having my code changed without any input on my side during a ci task.

it's not about being pro or against robot prettying your code. it's about applying automation carefully.

Currently beside, code being less readable (at least to me), it is inconsistent w.r.t. the style in which brackets are used. Which should not be the case, right?

**Comment by chicco785** [comment]
to make and example, now we have a combination of
```if () {
}
```
and
```if ()
{}
```

**Comment by jason-fox** [comment]
Adding consistent whitespace around the brackets is easy enough, but not what **prettier** is about really. You need a code formatter like [JsFormat](https://github.com/jdavisclark/JsFormat) for that.

What **prettier** did was to ensure that all `if` clauses had brackets e.g.:

```javascript
   if (x)
       doSomething();
```
becomes:

```javascript
   if (x)
       {doSomething();}
```
Which avoids problems like:

```javascript
   if (x)
       doSomething();
       doSomethingElse();
```

Where `doSomethingElse();` isn't in the `if` clause.

So why not improve with both **standard** syntax and **standard** formatting?

a. some-with-and-some-without-brackets . (Original)
b. all-with-brackets (Prettier)
c. all-with-brackets-and-nicely-indented (Prettier + JS Format)

**Comment by jason-fox** [comment]
Regarding failing the CI-build, all I have done is to amend the previous `eslint` as shown:

```yaml
       before_script:
        - eslint --ext .json .
        - jshint .
```

```yaml
       before_script:
        - npm run lint -C ./validator
```

The `lint` script is currently applying autofix via the `--fix` flag  this could of course be amended or split into two scripts or whatever.

File: validator/validate.js [hunk 36]
```diff
@@ -237,5 +257,6 @@ console.log('*** Warnings ***: ' +
 console.log('*** Errors ***: ' +
   JSON.stringify(msg.errors, null, '\t'));
 
-if (Object.keys(msg.errors).length != 0)
+if (Object.keys(msg.errors).length !== 0) {
   throw new Error(JSON.stringify(msg.errors, null, '\t'));
+}
```
Comment (chicco785) [inline validator/validate.js]: missing CR

File: validator/libs/checks.js (hunk 1 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js [hunk 37]
```diff
@@ -151,30 +170,31 @@ module.exports = {
...
       files = getFiles(fullPath + path.sep + '*-schema.json');
-    else
+    } else {
       files = getFiles('*-schema.json');
+    }
     debug("*loadLocalSchemas* - files: " + files);
     return files;
   }
-};
+};
```
Comment (chicco785) [inline validator/libs/schema.js]: missing cr

File: validator/libs/checks.js (hunk 2 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 3 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/message.js [hunk 38]
```diff
@@ -1,60 +1,63 @@
...
+  addValidExample(modelPath, message) {
     return addMessageToMap(modelPath, message, validExamples);
   },
   //add valid example to the valid examples map for a given model
-  addSupportedExample: function(modelPath, message) {
+  addSupportedExample(modelPath, message) {
     return addMessageToMap(modelPath, message, supportedExamples);
   }
-};
+};
```
Comment (chicco785) [inline validator/libs/message.js]: missing CR

File: validator/libs/checks.js (hunk 4 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 5 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/conf.js [hunk 39]
```diff
@@ -155,38 +153,39 @@ module.exports = {
...
+  validatePath() {
+    const stats = fs.lstatSync(nconf.get('dmv:path'));
     // Is it a directory?
     if (!stats.isDirectory()) {
       throw new Error('The path passed must be a directory');
     }
     return true;
   },
-};
+};
```
Comment (chicco785) [inline validator/libs/conf.js]: missing CR

File: validator/libs/checks.js (hunk 6 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js [hunk 40]
```diff
@@ -231,31 +253,34 @@ module.exports = {
...
+  idMatching(fullpath) {
     console.log('*** idMatching: not implemented ***');
   },
 
   //if a file matching a given regular expression exists in a given path
   // returns true, otherwise false
-  fileExists: fileExists
-};
+  fileExists
+};
```
Comment (chicco785) [inline validator/libs/checks.js]: missing CR

File: validator/libs/checks.js (hunk 7 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 8 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 9 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 10 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 11 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js [hunk 41]
```diff
@@ -71,119 +70,119 @@ var getApiEntityClient = function() {
...
+    let check = true;
     if (!fileExists(fullPath, 'README.md'))
-      check = false;
+      {check = false;}
     if (!check &&
         msg.addWarning(fullPath, 'does not include a Readme ' +
           'file README.md') && conf.failWarnings)
-      throw new Error('Fail on Warnings: ' +
-        JSON.stringify(msg.warnings, null, '\t'));
+      {throw new Error('Fail on Warnings: ' +
```
[the remark this answers is not in the document]
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

**Comment by chicco785** [comment]
as said, i would prefer no auto fix be applied by ci, but i am not stubborn about that :)

let's see @jmcanterafonseca feedback

File: validator/libs/checks.js (hunk 12 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 13 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 14 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 15 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/checks.js (hunk 16 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/message.js (hunk 17 above)
Comment (jason-fox) [reply validator/libs/message.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/message.js (hunk 18 above)
Comment (jason-fox) [reply validator/libs/message.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/message.js (hunk 19 above)
Comment (jason-fox) [reply validator/libs/message.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 20 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 21 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 22 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 23 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 24 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 25 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 26 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 27 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 28 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 31 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 32 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/libs/schema.js (hunk 33 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed with **JSFormatter** - cee624

File: validator/validate.js (hunk 36 above)
Comment (jason-fox) [reply validator/validate.js]: Fixed a2b27f09

File: validator/libs/schema.js (hunk 37 above)
Comment (jason-fox) [reply validator/libs/schema.js]: Fixed a2b27f09

File: validator/libs/message.js (hunk 38 above)
Comment (jason-fox) [reply validator/libs/message.js]: Fixed a2b27f09

File: validator/libs/conf.js (hunk 39 above)
Comment (jason-fox) [reply validator/libs/conf.js]: Fixed a2b27f09

File: validator/libs/checks.js (hunk 40 above)
Comment (jason-fox) [reply validator/libs/checks.js]: Fixed a2b27f09

**Comment by jason-fox** [comment]
With respect to whitespacing and brackets, adding a JavaScript **beautifier** should be kept as a separate PR. Obviously it will apply the same criteria over whether to `autofix` or not will apply for beautifying in the same manner as standardizing syntax.

**Comment by jason-fox** [comment]
@jmcanterafonseca  @chicco785 - maybe you would prefer that autofix is applied by `husky` on `git commit` rather than travis-ci.  I have altered the PR to use [lint-staged](https://github.com/okonet/lint-staged). 

* This has the advantage that no 💩  reaches the git repository

It does however do things to your code when you commit (which may be unexpected)

I've also separated `npm run lint` and `npm run prettier` into two separate commands for local use.

**Comment by chicco785** [comment]
it think it sounds better, because you will see changes locally before pushing to the remote repo, which also means that reverting is local and not remote, in case of need ;)

**Comment by jmcanterafonseca** [comment]
@chicco785 could you do a final check and LGTM it,  if suitable?

**Comment by chicco785** [comment]
tested locally, LGTM!

**Pull request merged**